commit ebb7e8e2201937ab31b3905093a0ee846ba8d566 (refs/changes/47/647/1)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-10-11T20:29:57+02:00 (3 years ago)
    
    TOOLS-1565 sdcadm post-setup ha-binder fails during zk data dir backup

diff --git a/lib/post-setup/ha-binder.js b/lib/post-setup/ha-binder.js
index 36d8da0..8443b17 100644
--- a/lib/post-setup/ha-binder.js
+++ b/lib/post-setup/ha-binder.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 /*
@@ -465,6 +465,16 @@ function do_ha_binder(subcmd, opts, args, cb) {
             }, next);
         },
 
+        function disableZkForBakup(_, next) {
+            self.progress('Disabling zookeeper for data backup');
+            shared.disableRemoteSvc({
+                server: vms[0].server_uuid,
+                zone: vms[0].uuid,
+                fmri: 'zookeeper',
+                log: self.log
+            }, next);
+        },
+
         function backupZookeeperData(_, next) {
             self.progress('Creating backup of zookeeper data directory ' +
                     '(this may take some time)');
@@ -488,6 +498,16 @@ function do_ha_binder(subcmd, opts, args, cb) {
             });
         },
 
+        function enableZkAfterBakup(_, next) {
+            self.progress('Enabling zookeeper after data backup');
+            shared.enableRemoteSvc({
+                server: vms[0].server_uuid,
+                zone: vms[0].uuid,
+                fmri: 'zookeeper',
+                log: self.log
+            }, next);
+        },
+
         function copyZkBackupToWorkDir(_, next) {
             self.progress('Copying backup of zookeeper data to: %s', wrkDir);
             var argv = [
